### ISSUE TITLE
Generating several bounding boxes per object

### ISSUE THREAD

**Issue by anvitmangal** [AlexeyAB/darknet#2673]
I was using YOLOv3 for detection of small objects and wanted to get several bounding boxes per object. I tried to reduce NMS while testing using the python darkness.py helpless but it didn't really help. Basically I wanted atleast 1 bounding box per image for py purposes. Is it possible to do so?

**Comment by AlexeyAB** [comment]
@anvitmangal Why do you want to get many False-Positives bounded boxes around 1 objects?

**Comment by anvitmangal** [comment]
@AlexeyAB I'm using YOLOv3 for lesion detections. For this, FP is not an issue but FN is not acceptable. For this reason I need possible bounding boxes in each test image, and with that I need the confidence scores for a particular class (Malignant). I tried doing so using the python helper darknet.py but when the lesion was detected as Benign, the Malignant confidence scores were coming out to be 0 when I printed them.

TLDR;
1) I need bounding boxes in each of my test images.
2) for each bounding box I need a non-zero malignant confidence score.

Note: I have trained the network for a binary class problem (Benign and Malignant)

**Comment by AlexeyAB** [comment]
@anvitmangal 

Just set `thresh = 0.01`
here: https://github.com/AlexeyAB/darknet/blob/1cd332e4cac001ffcc12a24c72640fe02b69a8a0/darknet.py#L294
or before this line: https://github.com/AlexeyAB/darknet/blob/1cd332e4cac001ffcc12a24c72640fe02b69a8a0/darknet.py#L255-L256

**Comment by anvitmangal** [comment]
@AlexeyAB thanks for this workaround. 
Still have to make something clear. So if a bounding box is predicted as Benign, why are the confidence scores for the other class, i.e., Malignant coming out to be 0 always. Shouldn't they be non-zero for some cases too?

**Comment by AlexeyAB** [comment]
> the Malignant confidence scores were coming out to be 0 when I printed them.

> Malignant coming out to be 0 always. Shouldn't they be non-zero for some cases too?

Can you show a screenshot?

**Comment by anvitmangal** [comment]
@AlexeyAB 
/Users/anvitmangal/Desktop/Screen Shot 2019-03-22 at 11.07.53 AM.png

**Comment by anvitmangal** [comment]
![Screen Shot 2019-03-22 at 11 07 53 AM](https://user-images.githubusercontent.com/23377962/54802893-456a2b00-4c93-11e9-8f8b-2f7d01ccc536.png)

**Comment by anvitmangal** [comment]
![Screen Shot 2019-03-22 at 11 07 53 AM](https://user-images.githubusercontent.com/23377962/54802916-5e72dc00-4c93-11e9-95b0-173cb22aeeb7.png)
I have printed out the labels in this fashion:
 finename, x, y, w, h, malignant confidence, Class label.

I got the malignant confidence by setting **dets[j].prob[_1_]** as class 0 is benign and class 1 is malignant.
